### ISSUE TITLE
[DesktopDateTimePicker] year start in 1900 instead of the selected/current year

### ISSUE THREAD

**Issue by Vishal-Kank** [mui/mui-x#6187]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1.click https://mui.com/x/react-date-pickers/date-time-picker/#responsiveness
2.select For Desktop date picker by clicking on its icon
3.click on year on top 

Demo of same
https://stackblitz.com/run?file=demo.js

### Current behavior 😯

When we click on year to change the year we can see the very start year 1900 we need to scroll down to see the current year.

### Expected behavior 🤔

When we click on year to change the year we should be able to see the current or already selected year.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by yaredtsy** [comment]
@Vishal-Kank  I have tried to solve it. if the main contributors approve it I will send a pull request.
Solution Links
-  demo:https://codesandbox.io/s/winter-forest-u2wnsq?file=/demo.js
-  git-code: https://github.com/yaredtsy/mui-x/pull/3/commits/5a9722a4fa40c60802465875ee4cabc075733a58

**Comment by alexfauquette** [comment]
We already did a fix in #6089 that is currently on `next` but could be included in master too.

**Comment by thibaultadet** [comment]
I have a similar problem when entering the date manually with `DatePicker`, the year is automatically set in 1900s.
For example : if i want 2022, I'll start with 2 and 1902 is set automatically.
I don't know if I should open a new issue for this... I believe a similar issue was fixed for the `DataGrid` component (#1920)

**Comment by alexfauquette** [comment]
@thibaultadet Yes this is unrelated to this issue which is about scrolling
